pkg/arvo/app/chat.hoon

```diff
@@ -60,161 +53,89 @@
+::  +prep: set up the app, migrate the state
 ::
 ++  prep
   |=  old=(unit state)
   ^-  (quip move _this)
-  =/  launcha/poke
-    [%launch-action [%chat /chattile '/~chat/js/tile.js']]
-  ?~  old
-    =/  inboxpat  /circle/inbox/config/group
-    =/  circlespat  /circles/[(scot %p our.bol)]
-    =/  inboxwir  /circle/[(scot %p our.bol)]/inbox/config/group
-    =/  inboxi/poke
-      :-  %hall-action
-      [%source %inbox %.y (silt [[our.bol %i] ~]~)]
-    =/  fakeannounce=poke
-      :-  %hall-action
-      [%create %hall-internal-announcements '' %village]
-    =/  announce=poke
-      :-  %hall-action
-      [%create %announcements 'Announcements from Tlon' %journal]
-    =/  help=poke
-      :-  %hall-action
-        [%create %urbit-help 'Get help about Urbit' %channel]
-    =/  dev=poke
-      :-  %hall-action
-      [%create %urbit-dev 'Chat about developing on Urbit' %channel]
-    =/  sourcefakeannounce/poke
-      :-  %hall-action
-      [%source %inbox %.y (silt [[our.bol %hall-internal-announcements] ~]~)]
-    =/  sourceannounce/poke
-      :-  %hall-action
-      [%source %inbox %.y (silt [[~marzod %announcements] ~]~)]
-    =/  hallactions=(list move)
+  ?^  old
+    :_  this(+<+ u.old)
+    [(launch-poke [/chattile '/~chat/js/tile.js'])]~
+  ::
+  =/  inbox-path  /circle/inbox/config/group
+  ::
+  :_  this
+  :*  [ost.bol %connect / [~ /'~chat'] %chat]
+      (launch-poke [/chattile '/~chat/js/tile.js'])
+      (hall-peer /circle/(scot %p our.bol)/inbox/config/group inbox-path)
+      (hall-peer /circles/(scot %p our.bol) /circles/(scot %p our.bol))
+      (hall-source [our.bol %i])
       ?:  =((clan:title our.bol) %czar)
         ~
       ?:  =(our.bol ~marzod)
-        ~&  %marzod-chat
-        :-  [ost.bol %poke /announce [our.bol %hall] announce]
-        [ost.bol %poke /announce [our.bol %hall] sourceannounce]~
+        :-  (hall-create %announcements 'Announcements from Tlon' %journal)  
+        [(hall-source [~marzod %announcements])]~
       ?:  =(our.bol ~dopzod)
-        ~&  %dopzod-chat
-        :-  [ost.bol %poke /announce [our.bol %hall] dev]
-        [ost.bol %poke /announce [our.bol %hall] help]~
-      :-  [ost.bol %poke /announce [our.bol %hall] fakeannounce]
-      :-  [ost.bol %poke /announce [our.bol %hall] sourcefakeannounce]
-      [ost.bol %poke /announce [our.bol %hall] sourceannounce]~
-    =/  moves=(list move)
-    :~  [ost.bol %peer inboxwir [our.bol %hall] inboxpat]
-        [ost.bol %peer circlespat [our.bol %hall] circlespat]
-        [ost.bol %connect / [~ /'~chat'] %chat]
-        [ost.bol %poke /chat [our.bol %hall] inboxi]
-        [ost.bol %poke /chat [our.bol %launch] launcha]
-    ==
-    :_  this
-    %+  weld  moves  hallactions
-  :-  [ost.bol %poke /chat [our.bol %launch] launcha]~
-  this(sta u.old)
+        :-  (hall-create %urbit-dev 'Chat about developing on Urbit' %channel)
+        [(hall-create %urbit-help 'Help about Urbit' %channel)]~
```

Comment:
Probably do a `:~` here also, like you do below.

Reply:
Ping.

Reply:
Over 80 chars? I meant the following, which adds zero extra width, and is more clearly a list.

```hoon
:~  (hall-create %urbit-dev 'Chat about developing on Urbit' %channel)
    (hall-create %urbit-help 'Help about Urbit' %channel)
==
```